master/internal/db/postgres_tasks_intg_test.go

```diff
@@ -25,6 +26,57 @@ import (
 	"github.com/determined-ai/determined/proto/pkg/taskv1"
 )
 
+// TestAddAllocationTimes checks that an allocation's start time is always < end time.
+// Allocations are allowed to be inserted/updated in to the table multiple times in
+// the event of a master restart -- check that the start & end times are set accordingly.
+func TestAddAllocationTimes(t *testing.T) {
+	// Set up the DB.
+	require.NoError(t, etc.SetRootPath(RootFromDB))
+	db := MustResolveTestPostgres(t)
+	MustMigrateTestPostgres(t, db, MigrationsFromDB)
+
+	// Set up the allocation.
+	mockUser := RequireMockUser(t, db)
+	mockTask := RequireMockTask(t, db, &mockUser.ID)
+	mockAllocation := model.Allocation{
+		AllocationID: model.AllocationID(fmt.Sprintf("%s-1", mockTask.TaskID)),
+		TaskID:       mockTask.TaskID,
+		StartTime:    ptrs.Ptr(time.Now().UTC()),
+	}
+
+	// First add an allocation.
+	err := db.AddAllocation(&mockAllocation)
+	require.NoError(t, err)
+
+	// Check that it's in the table.
+	_, err = db.AllocationByID(mockAllocation.AllocationID)
+	require.NoError(t, err)
+
+	// Update the allocation's end time manually.
+	_, err = Bun().NewUpdate().Table("allocations").
+		Where("allocation_id = ?", mockAllocation.AllocationID).
+		Set("end_time = ?", time.Now().UTC()).Exec(context.TODO())
+	require.NoError(t, err)
```

Comment:
nit: IIRC we have functions that will set the allocation's end_time ([CompleteAllocation](https://github.com/determined-ai/determined/blob/7c765ae9afa0228511c50081eeac2aefc4d13628/master/internal/db/postgres_tasks.go#L140)). Is there a reason you went with manually setting setting the end_time?

Reply:
good catch